### ISSUE TITLE
Handle multiple (>= 3) variance components?

### ISSUE THREAD

**Issue by mmkim1210** [OpenMendel/VarianceComponentModels.jl#17]
Thank you for the wonderful package. Things seem to run nicely. I have a couple questions. 

1. It appears that the package only handles up to two variance components, whether be it univariate or multivariate? Is there a plan to expand to handle more than 3 variance components?

2. What would be the appropriate way of doing inference for these variance components? I imagine for univariate case, you can fit a null model without the variance component of interest, and running some variation of likelihood ratio test as done in GCTA, but would you recommend a different way? Also for multivariate cases, where you have the covariance parameters, is there a way to easily fit the null model with such a term set to zero, and run LRT?

**Comment by Hua-Zhou** [comment]
These are two very good questions.

1. I have been wanting to expand the package to handle >=3 variance
components for long. Just hard to find a time to do it. If you are
interested, we can discuss how to proceed.

2. For the univariate response case, efficiently testing one or a subset of
variance components being zero among >= 3 variance components is still an
open question. In practice people resort to computationally intensive
methods such as bootstrap. The difficulty is (1) in genetics usually the
covariance matrix (GRM + identity) is not block diagonal so the usual
asymptotics does not apply and (2) the true paymaster lies on the boundary
so the usual  asymptotic null distribution does not apply. See the below
paper and references therein
http://hua-zhou.github.io/media/pdf/ZhouHuQiaoChoZhou16Boost.pdf
The multivariate response case is even harder.

In summary, testing when there are >= 3 variance components and testing
multivariate response variance component models are still open questions.
The LRT approach used in GCTA might work to some extent when parameter lies
in interior but I never saw theoretical justification of that approach.
